src/openrct2/util/util.h

```diff
@@ -56,4 +56,12 @@ uint32 util_rand();
 uint8 *util_zlib_deflate(uint8 *data, size_t data_in_size, size_t *data_out_size);
 uint8 *util_zlib_inflate(uint8 *data, size_t data_in_size, size_t *data_out_size);
 
+#define add_clamp(value, value_to_add, min_cap, max_cap) \
+	if ((max_cap - abs(value_to_add)) < abs(value) && (value < 0) == (value_to_add < 0)) { \
```

Comment:
@HaasJona I tested your code and it works, it also resolves the previous issue i mentioned having about overflowing negative numbers. It also still stays relatively small and only adds 2 more if checks (which is what i was hoping to keep as low as possible, since this code might get used alot for functions that get called often, such as finance_payment).. I updated this to use your code, thanks!

P.S. your code doesn't actually need the last end bracket, as this define doesn't have an opening bracket ;)